admin/controllers/school.js

```diff
@@ -449,14 +450,55 @@ const getSelectPupilNotTakingCheck = async (req, res, next) => {
 
   return res.render('school/select-pupils-not-taking-check', {
     breadcrumbs: req.breadcrumbs(),
+    sortField,
+    sortDirection,
     attendanceCodes,
-    formData,
     pupilsList,
     htmlSortDirection,
     arrowSortDirection
   })
 }
 
+const savePupilNotTakingCheck = async (req, res, next) => {
+  res.locals.pageTitle = 'Save pupils not taking the check'
+  req.breadcrumbs(res.locals.pageTitle)
+
+  if (req.body.attendanceCode === undefined || req.body.pupil === undefined) {
+    return res.redirect('/school/pupils-not-taking-check/select-pupils')
+  }
+
+  const todayDate = moment(moment.now()).format()
+  const pupilsData = await fetchMultiplePupils(Object.values(req.body.pupil))
+  let pupilsToSave = []
+
+  await pupilsData.map(async (pupil) => {
+    if (pupil) {
```

Comment:
Do we need this `if` condition?  Won't the result of `fetchMultiplePupils` either be an array of pupils or null?